### ISSUE TITLE
Look into all failing unit tests

### ISSUE THREAD

**Issue by hideki** [couchbase/couchbase-lite-java-core#1459]
Current failing test cases - 17 test cases.
- [x] ApiTest.failingTestMultiDocumentUpdateInTransaction()
- [x] DatabaseAttachmentTest.failingTestSetAttachmentsSequentiallyInTransaction
- [ ] DocumentTest.failingTestDocumentPropertiesAreImmutable <<-- NOT FIX?
- [x] DocumentTest.failingTestGetDocumentWithLargeJSON
- [ ] DocumentTest.failingTestProvidedMapChangesAreSafe - #1491
- [x] RouterTest.failingTestPostBulkDocsWithConflict
- [ ] ViewsTest.failingTestAllDocumentsLiveQuery <<-- NOT FIX?
- [ ] ViewsTest.failingTestJapaneseAsKey
- [ ] RemoteRequestTest.failingTestRetryQueueDeadlock
- [ ] ReplicationMockWebServerTest.failingTestContinuousPushReplicationGoesIdleTooSoon - #1077
- [ ] ReplicationMockWebServerTest.failingTestMockPullBulkDocsSyncGw - #1077
- [ ] ReplicationMockWebServerTest.failingTestMockPullerRestart  - #1490
- [ ] ReplicationMockWebServerTest.failingTestReplicationRestartPreservesValues  - #1490
- [ ] ReplicationMockWebServerTest.failingTestMockContinuousPullCouchDb  - #1490
- [ ] ReplicationMockWebServerTest.testGoOnlinePuller - #1494
- [x] ReplicationMockWebServerTest.failingTestRemoteConflictResolution
- [x] ReplicationMockWebServerTest.failingTestStopReplicatorWhenRetryingReplicationWithPermanentError

**Comment by hideki** [comment]
- DocumentTest.failingTestDocumentPropertiesAreImmutable

Question: Immutability of nested dictionaries might cause performance issue. I wonder if it worth to fix this.

```java
    public void failingTestDocumentPropertiesAreImmutable() throws Exception {
        String jsonString = "{\n" +
                "    \"name\":\"praying mantis\",\n" +
                "    \"wikipedia\":{\n" +
                "        \"behavior\":{\n" +
                "            \"style\":\"predatory\",\n" +
                "            \"attack\":\"ambush\"\n" +
                "        },\n" +
                "        \"evolution\":{\n" +
                "            \"ancestor\":\"proto-roaches\",\n" +
                "            \"cousin\":\"termite\"\n" +
                "        }       \n" +
                "    }   \n" +
                "\n" +
                "}";
        Map map = (Map) Manager.getObjectMapper().readValue(jsonString, Object.class);
        Document doc = createDocumentWithProperties(database, map);
        boolean firstLevelImmutable = false;
        Map<String, Object> props = doc.getProperties();
        try {
            props.put("name", "bug");
        } catch (UnsupportedOperationException e) {
            firstLevelImmutable = true;
        }
        assertTrue(firstLevelImmutable);

        boolean secondLevelImmutable = false;
        Map wikiProps = (Map) props.get("wikipedia");
        try {
            wikiProps.put("behavior", "unknown");
        } catch (UnsupportedOperationException e) {
            secondLevelImmutable = true;
        }
        assertTrue(secondLevelImmutable);

        boolean thirdLevelImmutable = false;
        Map evolutionProps = (Map) wikiProps.get("behavior");
        try {
            evolutionProps.put("movement", "flight");
        } catch (UnsupportedOperationException e) {
            thirdLevelImmutable = true;
        }
        assertTrue(thirdLevelImmutable);
    }
```

**Comment by hideki** [comment]
##### DocumentTest.failingTestProvidedMapChangesAreSafe

- Note: Document property should be copied. This issue should be Fixed.

```java
    public void failingTestProvidedMapChangesAreSafe() throws Exception {
        Map<String, Object> originalProps = new HashMap<String, Object>();
        Document doc = createDocumentWithProperties(database, originalProps);

        Map<String, Object> nestedProps = new HashMap<String, Object>();
        nestedProps.put("version", "original");
        UnsavedRevision rev = doc.createRevision();
        rev.getProperties().put("nested", nestedProps);
        rev.save();

        nestedProps.put("version", "changed");
        assertEquals("original", ((Map) doc.getProperty("nested")).get("version"));
    }
```

**Comment by hideki** [comment]
##### ReplicationMockWebServerTest.failingTestContinuousPushReplicationGoesIdleTooSoon

- Note: This test failure will be fixed when #1077 is fixed

**Comment by hideki** [comment]
##### Following three failure is caused by fixing 406 HTTP Error handling.
- ReplicationMockWebServerTest.failingTestMockPullerRestart 
- ReplicationMockWebServerTest.failingTestReplicationRestartPreservesValues
- ReplicationMockWebServerTest.failingTestMockContinuousPullCouchDb

Note: Above three failures should be fixed by fixing test code or MockWebServer
